test/apps/iodemo/io_demo.cc

```diff
@@ -153,33 +153,70 @@ class IoDemoRandom {
     template <typename T>
     static inline T rand(T min = std::numeric_limits<T>::min(),
                          T max = std::numeric_limits<T>::max() - 1) {
-        _seed = (_seed * _A + _C) & _M;
+        return rand(_seed, min, max);
+    }
+
+    template <typename T>
+    static inline T rand(unsigned &seed, T min = std::numeric_limits<T>::min(),
+                         T max = std::numeric_limits<T>::max() - 1) {
+        seed = (seed * _A + _C) & _M;
         /* To resolve that LCG returns alternating even/odd values */
         if (max - min == 1) {
-            return (_seed & 0x100) ? max : min;
+            return (seed & 0x100) ? max : min;
         } else {
-            return T(_seed) % (max - min + 1) + min;
+            return T(seed) % (max - min + 1) + min;
         }
     }
 
-    static inline void fill(void *buffer, size_t size) {
+    static inline void fill(unsigned &seed, void *buffer, size_t size) {
         size_t tail_count = size & sizeof(uint64_t);
         size_t body_count = (size - tail_count) / sizeof(uint64_t);
         uint64_t *body    = reinterpret_cast<uint64_t*>(buffer);
         uint8_t *tail     = reinterpret_cast<uint8_t*>(body + body_count);
 
-        fill(body, body_count);
-        fill(tail, tail_count);
+        fill(seed, body, body_count);
+        fill(seed, tail, tail_count);
+    }
+
+    static inline size_t validate(unsigned &seed, const void *buffer,
+                                  size_t size) {
+        size_t tail_count    = size & sizeof(uint64_t);
```

Comment:
what is this doing?

Reply:
round body to `sizeof(uint64_t)`, then check the rest by `sizeof(uint8_t)` to minimize number of `rand` calls

Reply:
fixed